### PR TITLE
[cw] different options for class voting

### PR DESCRIPTION
**Pull request by ghost**
This PR adds the _top_, _dist nolog_ and _dist log_ options described [here](http://wortschatz.informatik.uni-leipzig.de/~cbiemann/software/CW.html#_Toc135565171) to the cw implementation.
The new tests use the same graph and values from the linked page.

### CODE REVIEW

File: src/main/java/de/tudarmstadt/lt/wsi/WSI.java [hunk 1]
```diff
@@ -154,60 +156,43 @@ public void findSenseClusters(Writer writer) throws IOException {
 			monitor.reportProgress(processedNodes);
 		}
 	}
-	
+
 	@SuppressWarnings("static-access")
 	public static void main(String args[]) throws IOException {
 		CommandLineParser clParser = new BasicParser();
 		Options options = new Options();
-		options.addOption(OptionBuilder.withArgName("file")
```
Comment (johannessimon) [inline src/main/java/de/tudarmstadt/lt/wsi/WSI.java:162]: All the reformattings below, are they intentional or "by accident"?

Comment (johannessimon) [inline src/main/java/de/tudarmstadt/lt/wsi/WSI.java]: All the reformattings below, are they intentional or "by accident"?

File: src/main/java/de/tudarmstadt/lt/wsi/WSI.java [hunk 2]
```diff
@@ -231,11 +216,20 @@ public static void main(String args[]) throws IOException {
...
 		WSI cwd = null;
 		if (cl.getOptionValue("clustering").toLowerCase().equals("cw")) {
 			cwd = new WSI(graphWrapper, n, ClusteringAlgorithm.ChineseWhispers);
+			String cwOption = cl.getOptionValue("cwOption");
+			CW.Option option;
+			if ("DIST_LOG".equals(cwOption)) {
+				option = Option.DIST_LOG;
+			} else if ("DIST_NOLOG".equals(cwOption)) {
+				option = Option.DIST_NOLOG;
+			} else {
```
Comment (johannessimon) [inline src/main/java/de/tudarmstadt/lt/wsi/WSI.java]: Maybe we should print a help message and exit if none of the three options matches the provided parameter

**Comment by johannessimon** [comment]
Thanks for the patch! All in all looks very good, great that you also provided unit tests. :-) I commented on two small points.

File: src/main/java/de/tudarmstadt/lt/wsi/WSI.java (hunk 1 above)
Comment (ghost) [inline src/main/java/de/tudarmstadt/lt/wsi/WSI.java:162]: More or less by accident. I am using the Eclipse built in formatter profile.

File: src/main/java/de/tudarmstadt/lt/cw/global/CWGlobal.java [hunk 3]
```diff
@@ -47,6 +48,9 @@ public static void main(String args[]) throws IOException {
 				.withDescription(
 						"max. number of similar words to process for a given word (size of word subgraph to be clustered)")
 				.isRequired().create("N"));
+		options.addOption(OptionBuilder.withArgName("cwOption").hasArg()
+				.withDescription("CW weighting option:  \"TOP\"(default), \"DIST_LOG\", \"DIST_NOLOG\"")
```
Comment (alexanderpanchenko) [inline src/main/java/de/tudarmstadt/lt/cw/global/CWGlobal.java:52]: The names need some comments: what TOP, DIST_LOG and DIST_NOLOG means. E.g. "top sums over the neighbourhood’s classes, dist downgrades the influence of a neighbouring node by its degree (number of edges of node)". You can even add link to the example: http://wortschatz.informatik.uni-leipzig.de/~cbiemann/software/CW.html#_Toc135565171

File: src/test/java/de/tudarmstadt/lt/cw/global/CWGlobalTest.java [hunk 4]
```diff
@@ -13,6 +13,8 @@
 
 import org.junit.Test;
 
+import de.tudarmstadt.lt.cw.CW.Option;
```
Comment (alexanderpanchenko) [inline src/test/java/de/tudarmstadt/lt/cw/global/CWGlobalTest.java:16]: Can you please add here tests to other modes as well to make sure that other options also work with the global clustering mode?

File: src/main/java/de/tudarmstadt/lt/cw/CW.java [hunk 5]
```diff
@@ -75,8 +96,28 @@ protected void relabelNode(N node) {
...
+				weight = (float) (weight / (Math.log(neighbours + 1)));
+				break;
+			}
+			case DIST_NOLOG: {
+				float neightbours = count(graph.getNeighbors(sourceNode));
+				weight = weight / neightbours;
+				break;
+			}
+			case TOP:
+				// nothing to do, use the weight of the edge
```
Comment (alexanderpanchenko) [inline src/main/java/de/tudarmstadt/lt/cw/CW.java:114]: Just a matter of style, but I would rather call edge.getWeight() every time you need weight above (and here you would do also weight = edge.getWeight(). In this case, you do not need such a comment.

**Comment by alexanderpanchenko** [comment]
@johannessimon Thanks a lot for review! 

@haftsteind Sorry for delay w/ comments, I was on vacations. Looks good to me too. I will merge for now, you may want to address some my comments in the next pull request. Cheers.

**Pull request merged**